File: C++/Baekjoon/Problem9465.cpp
#include <iostream>

using namespace std;

int maxx(int a, int b) {
    if (a > b) return a;
    else return b;
}

int main() {
    int T;
    cin >> T;

    for (int testcase = 0; testcase < T; testcase++) {
        int dp[2][100001];
        int arr[2][100001];

        int n;
        cin >> n;
        for (int i = 0; i < 2; i++) {
            for (int j = 1; j <= n; j++) {
                cin >> arr[i][j];
            }
        }

        dp[0][0] = dp[1][0] = 0;
        dp[0][1] = arr[0][1];
        dp[1][1] = arr[1][1];

        for (int i = 2; i <= n; i++) {
            dp[0][i] = maxx(dp[1][i - 1], dp[1][i - 2]) + arr[0][i];
            dp[1][i] = maxx(dp[0][i - 1], dp[0][i - 2]) + arr[1][i];
        }
        cout << maxx(dp[1][n], dp[0][n]) << '\n';
    }
}

File: Python/Baekjoon/Problem9465.py
T = int(input())
for testcase in range(T):
    N = int(input())
    dp = [[0 for i in range(N + 1)] for j in range(2)]
    arr = [0 for i in range(2)]

    for i in range(2):
        arr[i] = list(map(int, input().split()))
        arr[i].insert(0, 0)

    dp[0][1] = arr[0][1]
    dp[1][1] = arr[1][1]

    for i in range(2, N + 1):
        dp[0][i] = max(dp[1][i - 1], dp[1][i - 2]) + arr[0][i]
        dp[1][i] = max(dp[0][i - 1], dp[0][i - 2]) + arr[1][i]

    print(max(dp[1][N], dp[0][N]))
